addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -292,6 +293,18 @@ var LinkDialog = Dialog.extend({
         const el = this.el.querySelector('[name="link_style_color"]:checked');
         $(this.buttonOptsCollapseEl).collapse(el && el.value ? 'show' : 'hide');
     },
+    /**
+     * @private
+     */
+    _isFromAnotherHostName: function (url) {
+        if (url.match(/^http[s]?/i)) {
+            return !url.startsWith(window.location.origin);
+        }
+        if (url.startsWith("/")) {
+            return false;
+        }
+        return !url.includes(window.location.hostname);
```

Comment:
Maybe there is something to do with the `URL` class instead of all this. Something like
```javascript
const urlObj = new URL(url, window.location.origin);
return (urlObj.origin === window.location.origin);
```
?

Reply:
I propose something [here](https://github.com/odoo/odoo/pull/64622/files#diff-9e62be92a417595b3f64828e768863d211a9362c960f4bdc754168c983faca14R299-R310) 

As I understand it, using the URL class is a heuristic too and not a general solution. Here, it does the job but I think we may find some corner cases where it'll fail to return what a human being will think is right.